File: apps/api/app/schemas/device.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from ..models import DeviceStatus


class DeviceCreate(BaseModel):
    organization_id: UUID
    hostname: str = Field(min_length=1, max_length=255)
    device_type: str = Field(min_length=1, max_length=100)
    operating_system: str = Field(min_length=1, max_length=200)
    ip_address: str | None = Field(default=None, max_length=45)


class DeviceRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    organization_id: UUID
    hostname: str
    device_type: str
    operating_system: str
    ip_address: str | None
    status: DeviceStatus
    last_seen_at: datetime | None
    created_at: datetime
    updated_at: datetime

File: apps/web/src/types/device.ts
/** Device inventory (management view: apps/api/app/schemas/device_management.py). */
export type Device = {
  id: string
  organization_id: string
  hostname: string
  device_type: string
  operating_system: string
  ip_address: string | null
  status: 'active' | 'inactive' | 'retired'
  last_seen_at: string | null
  created_at: string
  updated_at: string
}
